client/client.go

```diff
@@ -49,17 +50,14 @@ type Client struct {
 	Transport http.RoundTripper
 
 	authed bool
-	inited bool
-	initMu sync.Mutex
+	inited int32
 	authMu sync.Mutex
 	auth   *auth.Request
 }
 
-func (c *Client) init() error {
-	c.initMu.Lock()
-	defer c.initMu.Unlock()
-	if c.inited {
-		return nil
+func (c *Client) init() {
+	if !atomic.CompareAndSwapInt32(&c.inited, 0, 1) {
```

Comment:
You're right, I did a revert but the code wasn't right. Should be fixed 61932cf28ae20fe17fb56eb315a3b8a7afb2a703